GetStream/stream-chat-android
# Fix sorting channels by Channel::lastMessageAt

closes #3469 

### 🎯 Goal

Fix sorting channels by `Channel::lastMessageAt`.

### 🛠 Implementation details

Fix mapping `ChannelEntity` to `Channel`.

### 🧪 Testing

Apply the patch:

<details>
  <summary>Patch file to apply these changes for testing</summary>

```
Index: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/channel/list/viewmodel/ChannelListViewModel.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/channel/list/viewmodel/ChannelListViewModel.kt b/stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/channel/list/viewmodel/ChannelListViewModel.kt
--- a/stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/channel/list/viewmodel/ChannelListViewModel.kt	(revision 9dd71a2a5bfc06252e555d43d1c8377987303866)
+++ b/stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/channel/list/viewmodel/ChannelListViewModel.kt	(date 1651752014462)
@@ -411,7 +411,7 @@
          * The default sorting option for queries.
          */
         @JvmField
-        public val DEFAULT_SORT: QuerySort<Channel> = QuerySort.desc("last_updated")
+        public val DEFAULT_SORT: QuerySort<Channel> = QuerySort.desc("last_message_at")
 
         /**
          *  The initial state.

```

</details>

1. Open the sample app and go to the first channel
2. Enable airplane mode  and send a message
3. Kill the app and reopen it with airplane mode enabled
4. The channel should be placed at top 

### ☑️Contributor Checklist

#### General

#### Code & documentation

### ☑️Reviewer Checklist
